### ISSUE TITLE
WebAssembly tests

### ISSUE THREAD

**Issue by choikwa** [kripken/Massive#21]
I realize this is mainly for asm.js, but .js containing wasm could be interesting. I could not build equivalent .js that uses wasm as I don't know the testing parameters (ie, I could build Box2d lib and testbed, but no actual test command) Are there plans to test wasm?

**Comment by kripken** [comment]
Heh, I just started to work on that today :) So yes, there are plans.

My idea is to
1. Update the tests here with new builds, with asm.js separated out.
2. Then include wasm builds too, at which point each test can easily be flipped between asm.js and wasm.

**Comment by choikwa** [comment]
Thanks for the quick response! I look forward to it.

**Comment by kripken** [comment]
Might as well reopen this to track the feature.

After some initial work, there is a beta version up [here](http://kripken.github.io/Massive/beta/), and in the `wasm` branch, with wasm tests parallel to the asm.js. Feedback, PRs, etc. are welcome of course.

Currently things seem to work, but probably need a lot more polish. One issue is that if the browser lacks wasm support, it just hits an error. Not sure what we should do there (refuse to run? run and show a modified score?).

**Comment by choikwa** [comment]
Tested with Firefox Nightly and it works. Wasm tests slightly faster than asmjs on throughput but doing really bad on preparation.

**Comment by kripken** [comment]
Yeah, I see the same. The slight speedup is probably wasm compiling faster, and compile time is a factor in these (small, but still). The preparation slowdown is asm.js has caching in firefox, but it looks like wasm doesn't yet.

**Comment by arunetm-zz** [comment]
@kripken do you have plans to refresh the massive - wasm benchmarks?

**Comment by kripken** [comment]
A simple refresh isn't a good option, wasm now has async compilation forced (some browsers won't load it otherwise), and so we would need to figure out a proper way to measure asm/wasm startup speeds.

Or, we could remove the startup portion, or we could remove the asm.js portion. But even with just wasm, startup is tricky since browsers do it differently.

Bottom line, this would take some thinking and design work here. Not sure if I have time myself.

**Comment by arunetm-zz** [comment]
Thank you!
